vic/drivers/shared_image/include/vic_nc_log.h

```diff
@@ -0,0 +1,70 @@
+/******************************************************************************
+ *
+ * @section DESCRIPTION
+ *
+ * Logging macros for netcdf routines
+ *
+ * @section LICENSE
+ *
+ * Copyright (c) 2010, Zed A. Shaw and Mongrel2 Project Contributors.
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are
+ * met:
+ *
+ *     * Redistributions of source code must retain the above copyright
+ *       notice, this list of conditions and the following disclaimer.
+ *
+ *     * Redistributions in binary form must reproduce the above copyright
+ *       notice, this list of conditions and the following disclaimer in the
+ *       documentation and/or other materials provided with the distribution.
+ *
+ *     * Neither the name of the Mongrel2 Project, Zed A. Shaw, nor the names
+ *       of its contributors may be used to endorse or promote products
+ *       derived from this software without specific prior written
+ *       permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS
+ * IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
+ * THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+ * PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR
+ * CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+ * EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+ * PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+ * PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+ * LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+ * NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *****************************************************************************/
+
+#ifndef __vic_nc_log_h__
+#define __vic_nc_log_h__
+
+#include <vic_def.h>
+
+// Macros for logging
+#define clean_ncerrno(e) (e == 0 ? "None" : nc_strerror(e))
+
+// Error Level is always active
+#ifdef NO_LINENOS
+#define log_ncerr(e, M, ...) print_trace(); fprintf(LOG_DEST, \
+                                                    "[ERROR] errno: %s: " M "\n", \
+                                                    clean_ncerrno(e), \
+                                                    ## __VA_ARGS__); \
+    exit(EXIT_FAILURE);
+#else
+#define log_ncerr(e, M, ...) print_trace(); fprintf(LOG_DEST, \
+                                                    "[ERROR] %s:%d: errno: %s: " M " \n", \
+                                                    __FILE__, __LINE__, \
+                                                    clean_ncerrno(e), \
+                                                    ## __VA_ARGS__); \
+    exit(EXIT_FAILURE);
+#endif
+
+#define check_nc_status(A, M, ...) if (A != NC_NOERR) {log_ncerr(A, M, \
```

Comment:
Why do you use `A` here instead of `e`? It implies a different use case (maybe it is, but in that case just explain briefly in a comment what the difference is between `A` and `e`.